app/src/org/commcare/activities/CommCareHomeActivity.java

```diff
@@ -158,10 +164,24 @@
 
     private boolean sessionNavigationProceedingAfterOnResume;
 
+    //REMINDER STUFF
+    private CommCareReminderService mBoundService;
+    private ServiceConnection mConnection;
+    private Object serviceLock = new Object();
+    boolean mIsBound = false;
+    boolean mIsBinding = false;
+    private Vector<Alert> allAlerts = new Vector<>();
+
     @Override
     protected void onCreateSessionSafe(Bundle savedInstanceState) {
```

Comment:
I'm not super familiar with this entry point, is its lifecycle cleanly bound to the notion of a user login/logout event in a way that is reliable? I'm a bit surprised to see things firing from here v. somewhere in the CommCareSessionService or somewhere less tied to user navigation